doc/R-vignette/igraph_extended_tutorial_R_vignette_english.Rmd

```diff
@@ -0,0 +1,682 @@
+
+```{r}
+g <- make_empty_graph()
+
+g <- add_vertices(g,3)
+```
+
+Similarly, to add vertices use `add_edges`:
+
+```{r}
+g <- add_edges(g, edges = c(1,2, 1,3))
+```
+
+Edges are added by specifying the source and target vertex for each edge. This call added two edges, one connecting vertices `1` and `2`, and one connecting vertices `1` and `3`. Edges are also numbered up from one (the *edge ID*) and have an optional name.
+
+**Warning**
+
+Creating an empty graph and adding vertices and edges as shown here can be much slower than creating a graph with its vertices and edges as demonstrated earlier. If speed is of concern, you should especially avoid adding vertices and edges *one at a time*. If you need to do it anyway, you can use `add_vertex` and `add_edge`.
+
+If you try to add edges to vertices with invalid IDs (i.e., you try to add an edge to vertex `5` when the graph has only three vertices), you get an error:
+
+```{r echo = TRUE, eval = FALSE}
+g <- add_edges(g, edges = c(5,4))
+```
+
+Let us add some more vertices and edges to our graph, in igraph we can usea the `magrittr` package who provides a mechanism for chasing commands with the operator %>%:
+
+```{r echo = TRUE}
+g <- g %>% add_edges(edges=c(3,1)) %>% add_vertices(3) %>% add_edges(edges=c(3,4, 4,5, 5,6, 6,4))
+g
+```
+
+We now have an directed graph with 6 vertices and 7 edges. Vertex and edge IDs are always *contiguous*, so if you delete a vertex all subsequent vertices will be renumbered. When a vertex is renumbered, edges are **not** renumbered, but their source and target vertices will. Use `delete_vertices` and `delete_edges` to perform these operations. For instance, to delete the edge connecting vertices `3-4`, get its ID and then delete it:
+
+```{r echo = TRUE}
+get.edge.ids(g, c(3,4))
+```
+
+```{r}
+g <- delete_edges(g, 4)
+```
+
+Another forms to delete edges:
+
+```{r echo = TRUE}
+g <- make_ring(10) %>% delete_edges("10|1")
+plot(g)
+```
+
+```{r echo = TRUE}
+g <- make_ring(5)
+g <- delete_edges(g, get.edge.ids(g, c(1,5, 4,5)))
+plot(g)
+```
+
+Let's elaborate another example. We have the task of elaborating a chordal graph. A graph is chordal (or triangulated) if each of its cycles of four or more nodes has a chord, which
+is an edge joining two nodes that are not adjacent in the cycle. Our initial graph is:
+
+```{r}
+g1 <- graph_from_literal(A-B:C:I, B-A:C:D, C-A:B:E:H, D-B:E:F,
+                         E-C:D:F:H, F-D:E:G, G-F:H, H-C:E:G:I,
+                         I-A:H)
+plot(g1)
+```
+
+The ‘:’ operator can be used to define vertex sets. If an edge operator connects two vertex sets then every vertex from the first set will be connected to every vertex in the second set. Then we use `is_chordal` to evaluate if our graph is chordal and to search what edges are missing to fill-in the graph:
+
+```{r echo = TRUE}
+is_chordal(g1, fillin=TRUE)
+```
+
+```{r echo = TRUE}
+chordal_graph <- add_edges(g1, c(2,6, 8,7, 5,7, 2,7, 6,1, 7,1))
+plot(chordal_graph)
+```
```

Comment:
Here you are copying the fillin manually, without mentioning where it came from. Why don't you use the result of `is_chordal(g1, fillin = T)$fillin` directly?

Reply:
thank you, done